### ISSUE TITLE
1.4.0 AppImage fails to start

### ISSUE THREAD

**Issue by k-kirill** [eneshecan/whatsapp-for-linux#160]
**Description**
Whatsapp 1.4.0 AppImage fails to start
```
sh: /tmp/.mount_Whatsa4piHWs/lib/x86_64-linux-gnu/libreadline.so.8: no version information available (required by sh)
<13>Jan 21 23:42:12 whatsapp-for-linux[7682]: 
<13>Jan 21 23:42:13 whatsapp-for-linux[7682]: ** (whatsapp-for-linux:7676): ERROR **: 23:42:13.184: Unable to spawn a new child process: Failed to execute child process “/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/WebKitNetworkProcess” (No such file or directory)
Trace/breakpoint trap (core dumped)
```

**To Reproduce**
1. Start Whatsapp 1.4.0 AppImage

**Environment**
 - openSUSE Tumbleweed
 - whatsapp AppImage 1.4.0
 - Gnome 41.3

**Comment by k-kirill** [comment]
Looks like the `/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/WebKitNetworkProcess` file is already in the AppImage 1.4.0. So eneshecan/whatsapp-for-linux@cb62703 will not help :(
```
$ ./whatsapp-for-linux-1.4.0-x86_64.AppImage --appimage-mount
/tmp/.mount_whatsaw7zDQo
```
```
$ ls -l /tmp/.mount_whatsaw7zDQo/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/WebKitNetworkProcess
-rwxr-xr-x 1 root root 6424 Sep 27 12:23 /tmp/.mount_whatsaw7zDQo/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/WebKitNetworkProcess
```

**Comment by eneshecan** [comment]
Unfortunately, I don't see that those files are copied into the appimage. Let's see if my latest commit fixes it...

**Comment by eneshecan** [comment]
Strange, if I build the appimage locally on my Ubuntu VM, it works fine. There are maybe some missing pieces in GitHub workflow...

**Comment by k-kirill** [comment]
From tauri-apps/tauri#2940
> the paths of those files are hardcoded into the libwebkit*.so file (hardcoded paths are a big no-no in AppImages).
I tried to understand the applied fix there, but i haven't been able to as it seems to be too hacky :(

**Comment by eneshecan** [comment]
> 

Thanks a lot, this is most probably the same issue. We can do something like this in our appimage recipe but I am pretty tired of packaging issues for now. If someone could take a look, would be nice so that we can provide this in the next release.

**Comment by k-kirill** [comment]
For the record, I reported the issue upstream: AppImageCrafters/appimage-builder#175. Not sure, if it helps or not.

**Comment by eneshecan** [comment]
> For the record, I reported the issue upstream: AppImageCrafters/appimage-builder#175. Not sure, if it helps or not.

Thanks a lot! It definitely helps.

**Comment by FabianLars** [comment]
@k-kirill 
> I tried to understand the applied fix there, but i haven't been able to as it seems to be too hacky :(

Yeah it really is bad/hacky, like it always is when you have to resort to manual binary patching. In theory this is something that needs to be fixed/changed in webkitgtk itself :/

The problem is that the paths to these files are hardcoded inside the libwebkit so file, pointing to /usr/lib*/... This raises 2 problems:
1) Obviously we can't rely on webkit being installed
2) Not every distro has those files in the same path

So the workaround (at least for us @tauri, see below for a possible alternative) is to "binary patch" these absolute paths to relative ones == replacing `/usr` with `././`
-> both values _need_ to be of the same length, therefore the double `./`
-> Here's how we do it: [tauri-apps/linuxdeploy-plugin-gtk](https://github.com/tauri-apps/linuxdeploy-plugin-gtk/blob/f0cbf6313aa9a18ba68a609908188ed6f75852c5/linuxdeploy-plugin-gtk.sh#L330) - and yes that's super ugly.

Soo, idk much about appimage-builder, but i guess this will be labeled as `won't fix` over there (it's not really their "responsibility") and you need to do it yourself, either with the feature introduced [here](https://github.com/AppImageCrafters/appimage-builder/issues/50#issuecomment-714853858) if you need to resort to binary patching too or with the **path_mappings** property if that works for you.

**Comment by eneshecan** [comment]
@FabianLars Thanks a lot for the detailed information. I think we should apply the fix/hack as you did for `tauri`.
